frontend/src/admin/use-surveys.ts

```diff
@@ -5,13 +5,19 @@ import {Survey, surveysFromJson} from "zero-zummon"
 type UseSurveyReturn = {
     loading: boolean,
     surveys: Survey[],
+    // for syncing the state
+    changeSurvey: (newSurvey: Survey) => void,
     removeSurvey: (surveyId: string) => void,
 }
 
 export const useSurveys = (): UseSurveyReturn => {
     const [loading, setLoading] = useState(true)
     const [surveys, setSurveys] = useState<Survey[]>([])
 
+    const changeSurvey = (newSurvey: Survey) => {
+        setSurveys(surveys.map(s => s.id.toString() === newSurvey.id.toString() ? newSurvey : s))
```

Comment:
use 'survey' instead of 's', it's clear but it can be more specific :P

Reply:
I blame copilot. Done.